UnityProject/Assets/Scripts/Items/Tool/Pickaxe.cs

```diff
@@ -25,40 +29,67 @@ public bool WillInteract(PositionalHandApply interaction, NetworkSide side)
 
 	public void ServerPerformInteraction(PositionalHandApply interaction)
 	{
-		//server is performing server-side logic for the interaction
-		//do the mining
-		void ProgressComplete() => FinishMine(interaction);
 
-		//Start the progress bar:
-		//technically pickaxe is deconstruction, so it would interrupt any construction / deconstruction being done
-		//on that tile
-		//TODO: Refactor this to use ToolUtils once that's merged in
+		var interactableTiles = interaction.TargetObject.GetComponent<InteractableTiles>();
+		var wallTile = interactableTiles.MetaTileMap.GetTileAtWorldPos(interaction.WorldPositionTarget, LayerType.Walls) as BasicTile;
+		var calculatedMineTime = wallTile.MiningTime * timeMultiplier;
+		positionalHandApply = interaction;
 
-		var bar = StandardProgressAction.Create(ProgressConfig, ProgressComplete)
-			.ServerStartProgress(interaction.WorldPositionTarget.RoundToInt(),
-				5f, interaction.Performer);
-		if (bar != null)
-		{
-			SoundManager.PlayNetworkedAtPos(pickaxeSound, interaction.WorldPositionTarget, sourceObj: interaction.Performer);
-		}
+		ToolUtils.ServerUseToolWithActionMessages(
+			interaction, calculatedMineTime,
+			$"You start mining the {objectName}...",
```

Comment:
`objectName` isn't set. It _does_ get set in the other ServerPerformInteraction, however.

Reply:
While you're at it, consider rescoping the `objectName` variables to within those methods; don't think it needs to be a class field.

Reply:
>  however. You should be able to get the name of the tile from the interaction

TargetObject is null for some reason which is preventing the tile region to trigger. Also you can't get the tile name from the interaction afaik, there is a ``DisplayName`` field in ``BasicTile`` but I rarely see it filled.

Reply:
Right, tiles aren't objects. This is `PositionalHandApply`, you're probably after `TileApply`.